### PR TITLE
add new rules for storage account (azure-IaC)

### PR DESCRIPTION
**Pull request by shahinshirvani**

### CODE REVIEW

File: azure/iac/storageaccounts.rego [hunk 1]
```diff
@@ -726,3 +726,348 @@ storage_account_latest_tls_configured_metadata := {
...
+
+azure_attribute_absence["storage_account_private_endpoint"] {
+    count([c | lower(input.resources[_].type) == "microsoft.storage/storageaccounts"; c := 1]) != count([c | lower(input.resources[_].type) == "microsoft.network/privateendpoints"; c := 1])
+}
+
+azure_issue["storage_account_private_endpoint"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.network/privateendpoints"
+    privateLinkServiceConnection := resource.properties.privateLinkServiceConnections[_]
+    contains(lower(privateLinkServiceConnection.properties.privateLinkServiceId), "microsoft.storage/storageaccounts")
```
Comment (rezoan) [inline azure/iac/storageaccounts.rego:742]: @shahinshirvani either add a `not `before `contains `or rename `azure_issue `to `no_azure_issue `for readability purpose.

File: azure/iac/storageaccounts.rego [hunk 2]
```diff
@@ -726,3 +726,348 @@ storage_account_latest_tls_configured_metadata := {
...
+    "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts"
+}
+
+
+#
+# PR-AZR-ARM-STR-020
+#
+
+default storage_account_require_encryption = null
+
```
Comment (rezoan) [inline azure/iac/storageaccounts.rego:797]: @shahinshirvani add an attribute checking for `resource.properties.encryption`

**Review by rezoan — CHANGES_REQUESTED** [review]

File: azure/iac/storageaccounts.rego (hunk 2 above)
Comment (shahinshirvani) [reply azure/iac/storageaccounts.rego:797]: ```not obj.foo.bar.baz``` checks if path foo.bar.baz, foo.bar, or foo does not exist or is false

Re:https://www.openpolicyagent.org/docs/latest/policy-reference/

File: azure/iac/storageaccounts.rego [hunk 3]
```diff
@@ -726,3 +726,348 @@ storage_account_latest_tls_configured_metadata := {
...
+    "Policy Title": "Storage accounts should have infrastructure encryption",
+    "Policy Description": "Enable infrastructure encryption for higher level of assurance that the data is secure. When infrastructure encryption is enabled, data in a storage account is encrypted twice.",
+    "Resource Type": "microsoft.storage/storageaccounts",
+    "Policy Help URL": "",
+    "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts"
+}
+
+#
+# PR-AZR-ARM-STR-021
+#
```
Comment (rezoan) [inline azure/iac/storageaccounts.rego:865]: @shahinshirvani `PR-AZR-ARM-STR-020` and `PR-AZR-ARM-STR-021` are same rule. remove duplicate one.
See my comment at https://github.com/prancer-io/prancer-compliance-test/issues/343#issuecomment-991630268 and update accordingly.

**Review by rezoan — CHANGES_REQUESTED** [review]
@shahinshirvani please see my comment at https://github.com/prancer-io/prancer-compliance-test/issues/343#issuecomment-991630268 and remove duplicate rules.

File: azure/iac/storageaccounts.rego (hunk 3 above)
Comment (shahinshirvani) [reply azure/iac/storageaccounts.rego:865]: they are different as I understand
the ```PR-AZR-ARM-STR-021``` checks ```requireInfrastructureEncryption``` for this resource ```microsoft.storage/storageaccounts/encryptionscopes``` but ```PR-AZR-ARM-STR-020``` checks this resource ```microsoft.storage/storageaccounts```
both have same policy but different rules

**Review by rezoan — APPROVED** [review]

**Pull request merged**